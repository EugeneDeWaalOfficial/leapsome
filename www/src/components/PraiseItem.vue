<template>
  <div class="praise-item">
    <b-avatar
      :text="praise.sender.firstname[0] + praise.sender.lastname[0]"
      variant="secondary"
      size="5rem"
    ></b-avatar>
    <b-icon icon="caret-right-fill" scale="2" class="caret"></b-icon>
    <b-avatar
      :text="praise.receiver.firstname[0] + praise.receiver.lastname[0]"
      variant="secondary"
      size="5rem"
    ></b-avatar>

    <br /><br />

    <p>From {{ praise.sender.firstname }} to {{ praise.receiver.firstname }}</p>

    <hr />
    <p>{{ praise.content }}</p>
    <hr />
  </div>
</template>

<script>
export default {
  name: "PraiseItem",
  props: {
    praise: {
      type: Object,
      required: true
    }
  },
  data() {
    return {};
  }
};
</script>

<style scoped>
.praise-item {
  margin-bottom: 2rem;
}

.caret {
  margin: auto 1rem;
}
</style>
