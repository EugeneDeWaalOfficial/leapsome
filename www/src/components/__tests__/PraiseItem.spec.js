import { shallowMount } from "@vue/test-utils";

import PraiseItem from "@/components/PraiseItem.vue";

describe("PraiseItem.vue", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(PraiseItem, {
      propsData: {
        praise: {
          _id: "1",
          sender: { _id: "11", firstname: "Billie", lastname: "Some" },
          receiver: { _id: "12", firstname: "Lee", lastname: "Leaper" },
          content: "Great job!"
        }
      }
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
