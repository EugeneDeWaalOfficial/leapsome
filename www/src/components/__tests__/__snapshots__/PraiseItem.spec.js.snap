// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PraiseItem.vue renders correctly 1`] = `
<div
  class="praise-item"
>
  <b-avatar
    size="5rem"
    text="BS"
    variant="secondary"
  />
   
  <b-icon
    class="caret"
    icon="caret-right-fill"
    scale="2"
  />
   
  <b-avatar
    size="5rem"
    text="LL"
    variant="secondary"
  />
   
  <br />
  <br />
   
  <p>
    From Billie to Lee
  </p>
   
  <hr />
   
  <p>
    Great job!
  </p>
   
  <hr />
</div>
`;
